### DISCUSSION TITLE
Contributing through by adding more tests

### DISCUSSION THREAD

**Discussion by Sonichigo** [pocketbase/pocketbase#3808] [Q&A]
I was going through the codebase and pr, to see how ways to contribute to the pocketbase. I noticed that there are tests written but couldn't see more of integrations and e2e. Can i contirbute by adding more of these, I use a Keploy for generating testcases and data-mocks?

**Comment by ganigeorgiev** [comment]
Thank you, but for now there are no plans for e2e tests for the UI as it will be hard to maintain while still making UI changes with each release.

The Admin UI also most likely will be restructured sometime in the future to make it extendable (see https://github.com/pocketbase/pocketbase/issues/898).

**Reply by ganigeorgiev** [reply]
If you mean e2e tests for the server-side, then we already have a mixture of unit and integration tests (usually suffixed with `_test.go`; _note that the test files are located in the same directory with the related code_). The tests operate with the test database available in https://github.com/pocketbase/pocketbase/tree/master/tests/data.

If you think that it can be improved (eg. we currently don't have extensive integration tests for the realtime APIs) feel free to open a PR and I'll try to review it, but I don't like the idea to depend on an external tool for this.

**Reply by Sonichigo** [reply]
Yeah, I meant e2e tests for the server side. by realtime APIs you mean websockets are being used or HTTP ? Thank in advance for clarifying !!

**Reply by ganigeorgiev** [reply]
PocketBase realtime subscriptions uses SSE.

For the APIs tests you can explore the files in the [`apis` package](https://github.com/pocketbase/pocketbase/tree/master/apis).
